src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURE = 5;
+    private static final int MAX_RANDOM_NUMBER = 10;
+    public final Random random = new Random();
+    /*public final int randomNumber = random.nextInt(MAX_RANDOM_NUMBER);*/
+    public final ColorSupplier colorSupplier = new ColorSupplier();
+    /*private final String figureColor = colorSupplier.getRandomColor();*/
+    public static Color white = Color.WHITE;
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(NUMBER_OF_FIGURE);
+        /*int randomNumber = random.nextInt(MAX_RANDOM_NUMBER);
+        final String figureColor = colorSupplier.getRandomColor();*/
+        switch (figureType) {
+            case 0 -> {
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(MAX_RANDOM_NUMBER));

```

Comment:
same for other figures
```suggestion
                return new Square(randomColor, random.nextInt(MAX_RANDOM_NUMBER));
```